lib/semantics/mod-file.cc

```diff
@@ -690,28 +740,33 @@ Scope *ModFileReader::Read(const SourceName &name, Scope *ancestor) {
       return it->second->scope();
     }
   }
-  auto path{FindModFile(name, ancestorName)};
-  if (!path.has_value()) {
-    return nullptr;
-  }
-  // TODO: We are reading the file once to verify the checksum and then again
-  // to parse. Do it only reading the file once.
-  if (!VerifyHeader(*path)) {
-    context_.Say(name,
-        "Module file for '%s' has invalid checksum: %s"_err_en_US, name, *path);
-    return nullptr;
-  }
   parser::Parsing parsing{context_.allSources()};
   parser::Options options;
   options.isModuleFile = true;
   options.features.Enable(parser::LanguageFeature::BackslashEscapes);
-  parsing.Prescan(*path, options);
+  options.searchDirectories = context_.searchDirectories();
+  auto path{ModFileName(name, ancestorName, context_.moduleFileSuffix())};
+  parsing.Prescan(path, options);
+  const auto *sourceFile{parsing.sourceFile()};
```

Comment:
As mentioned above, if this is the only use of the new data member in `Parsing`, it would be a cleaner interface to return the pointer from `Prescan()` instead.

Reply:
Done